src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.cpp

```diff
@@ -47,6 +47,24 @@ HydroMaxTimeSeriesReader::HydroMaxTimeSeriesReader(PartHydro& hydro,
     dailyMaxPumpAndGen.reset(4U, DAYS_PER_YEAR, true);
 }
 
+static bool checkPower(const Matrix<>& dailyMaxPumpAndGen, const std::string& areaName)
+{
+    for (uint i = 0; i < 4U; ++i)
+    {
+        auto& col = dailyMaxPumpAndGen[i];
+        for (uint day = 0; day < DAYS_PER_YEAR; ++day)
+        {
+            if (col[day] < 0. || (i % 2U /*column hours*/ && col[day] > 24.))
+            {
+                logs.error() << areaName << ": invalid power or energy value";
+                return false;
```

Comment:
Carefull
With the early return you are returning after the first error. Previous behaviour is to check all values and log all errors.

Reply:
if (!errorPower  avoided going into this scope after the first error, I simplified while keeping the same behavior

Reply:
No you don't keep the same behavior. The user will only see the 1st error, not the following ones if any.